modules/common/util/env_test.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2023 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestGetEnvVar(t *testing.T) {
+
+	envVarName := fmt.Sprintf("%sTEST", t.Name())
+	envVarVal := "testing"
+	envVarValDefault := "default"
+
+	t.Setenv(envVarName, envVarVal)
+
+	tests := []struct {
+		name string
+		data []string
+		want string
+	}{
+		{
+			name: "Get env var where it is actually present and there is a default",
+			data: []string{envVarName, envVarValDefault},
+			want: envVarVal,
+		},
+		{
+			name: "Get env var where it is actually present and there is no default",
+			data: []string{envVarName, ""},
+			want: envVarVal,
+		},
+		{
+			name: "Get env var where it is not actually present and there is a default",
+			data: []string{"some_absent_env_var", envVarValDefault},
+			want: envVarValDefault,
+		},
+		{
+			name: "Get env var where it is not actually present and there is no default",
+			data: []string{"some_absent_env_var", ""},
+			want: "",
```

Comment:
What's the use case for GetEnvVar ? could also return an err for this case. I guess the caller needs to validate for empty string now?

Reply:
I think the use case is coming from here https://github.com/openstack-k8s-operators/nova-operator/pull/335/files#diff-8731911f4dc06ba38aa36b07378ab68d4252b6a44549bb10296e1f48d9687180R164-R172 

When the containerImage defaults are set up for the webhook we want to use the ENV value if the ENV is set, otherwise we want to use a hardcoded default value for the containerImage.

Reply:
Yes, the use case is in the context of defaults via webhooks as @gibizer mentioned.  We could have it return an error in certain circumstances, but we'd have to think about when to do so.  Would it be when `key` is not set in ENV and `baseDefault` was provided as `""`?

Maybe this should go in a separate `webhook` package so as to avoid confusion as to its purpose?

Reply:
I don't think we have to handle the case specially when baseDefault is provided as "" as that means the caller explicitly provided us "".

Reply:
Fair enough.  I will remove that test case.